"FATAL	Fatal error	report error: unable to write results: failed to create a file: failed to create output file" in 1.6.2+

**Describe the bug**
After the release of either 1.6.2 or 1.6.3 today, our Azure DevOps Pipeline fails with the following error

"FATAL	Fatal error	report error: unable to write results: failed to create a file: failed to create output file: open /home/vsts/work/_temp/trivy-results-0.7459715129666153.json: no such file or directory"

It was working as intended earlier today. Is this a regression or is there something in our configuration we should change?

**Trivy Azure Pipeline Task Verison**
The pipeline configuration we are running

![Image](https://github.com/user-attachments/assets/dfa44ec9-22c5-4e12-a0ee-724db714b560)

**Error message**

```
Starting: trivy
==============================================================================
Task         : Trivy
Description  : Trivy is the world’s most popular open source vulnerability and misconfiguration scanner. It is reliable, fast, extremely easy to use, and it works wherever you need it.
Version      : 1.6.3
Author       : Aqua Security
Help         : https://github.com/aquasecurity/trivy-azure-pipelines-task
==============================================================================
Run requested using docker...
Configuring options for image scan...
/usr/bin/docker run --rm -v /home/vsts/.docker:/root/.docker -v /tmp:/tmp -v /var/run/docker.sock:/var/run/docker.sock -v /home/vsts/work/1/s:/src --workdir /src aquasec/trivy:latest fs --exit-code 1 --format json --output /home/vsts/work/_temp/trivy-results-0.7459715129666153.json --ignore-unfixed --db-repository ghcr.io/aquasecurity/trivy-db:2,public.ecr.aws/aquasecurity/trivy-db .
Unable to find image 'aquasec/trivy:latest' locally
latest: Pulling from aquasec/trivy
1d671f98de6b: Pulling fs layer
2c38dcf52ab2: Pulling fs layer
2167091a7879: Pulling fs layer
2167091a7879: Waiting
1d671f98de6b: Verifying Checksum
1d671f98de6b: Download complete
38a8310d387e: Verifying Checksum
38a8310d387e: Download complete
38a8310d387e: Pull complete
2167091a7879: Verifying Checksum
2167091a7879: Download complete
2c38dcf52ab2: Verifying Checksum
2c38dcf52ab2: Download complete
1d671f98de6b: Pull complete
2c38dcf52ab2: Pull complete
2167091a7879: Pull complete
Digest: sha256:029e990b328d149bf0a9ffe355919041e1f86192db2df47e217f8a36dd42ceac
Status: Downloaded newer image for aquasec/trivy:latest
2025-02-17T11:29:22Z	INFO	Adding schema version to the DB repository for backward compatibility	repository="public.ecr.aws/aquasecurity/trivy-db:2"
2025-02-17T11:29:22Z	INFO	[vulndb] Need to update DB
2025-02-17T11:29:22Z	INFO	[vulndb] Downloading vulnerability DB...
2025-02-17T11:29:22Z	INFO	[vulndb] Downloading artifact...	repo="ghcr.io/aquasecurity/trivy-db:2"
INFO	[vulndb] Artifact successfully downloaded	repo="ghcr.io/aquasecurity/trivy-db:2"
2025-02-17T11:29:25Z	INFO	[vuln] Vulnerability scanning is enabled
2025-02-17T11:29:25Z	INFO	[secret] Secret scanning is enabled
2025-02-17T11:29:25Z	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2025-02-17T11:29:25Z	INFO	[secret] Please see also https://aquasecurity.github.io/trivy/v0.59/docs/scanner/secret#recommendation for faster secret detection
2025-02-17T11:29:27Z	INFO	Suppressing dependencies for development and testing. To display them, try the '--include-dev-deps' flag.
2025-02-17T11:29:27Z	INFO	Number of language-specific files	num=1
2025-02-17T11:29:27Z	INFO	[poetry] Detecting vulnerabilities...
2025-02-17T11:29:27Z	FATAL	Fatal error	report error: unable to write results: failed to create a file: failed to create output file: open /home/vsts/work/_temp/trivy-results-0.7459715129666153.json: no such file or directory
##[error]Failed: Trivy detected problems.
##[error]Unable to process command '##vso[task.addattachment type=JSON_RESULT;name=trivy0.46455983181214644.json;]/home/vsts/work/_temp/trivy-results-0.7459715129666153.json' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Value cannot be null. (Parameter 'Cannot upload task attachment file, attachment file location is not specified or attachment file does not exist on disk.')
##[error]Trivy seems to have failed so no output path to generate reports from.
Done!
Finishing: trivy
```

--- a/Makefile
+++ b/Makefile
@@ -10,10 +10,19 @@ lint:
 	cd ui && npm install -f && npm run lint
 	cd trivy-task && npm install -f && npm run lint
 
+.PHONY: lint-fix
+lint-fix:
+	cd ui && npm install -f && npm run lint:fix
+	cd trivy-task && npm install -f && npm run lint:fix
+
 .PHONY: format
 format:
 	cd ui && npm install -f && npm run format
 	cd trivy-task && npm install -f && npm run format
+
+
+.PHONY: tidy
+tidy: lint-fix format
 
 .PHONY: build-ui
 build-ui: clean

--- a/trivy-task/index.ts
+++ b/trivy-task/index.ts
@@ -7,8 +7,12 @@ import { generateAdditionalReports } from './additionalReporting';
 
 async function run() {
   task.debug('Starting Trivy task...');
-  const outputPath = tmpPath + 'trivy-results-' + Math.random() + '.json';
-  task.rmRF(outputPath);
+  const resultsFile = `trivy-results-${Math.random()}.json`;
+  const localOutputPath = `${tmpPath}/${resultsFile}`;
+  const outputPath = task.getBoolInput('docker', false)
+    ? `/tmp/${resultsFile}`
+    : localOutputPath;
+  task.rmRF(localOutputPath);
   const scanPath = task.getInput('path', false);
   const image = task.getInput('image', false);
   const scanners = task.getInput('scanners', false) ?? '';
@@ -89,11 +93,7 @@ async function run() {
   }
 
   task.debug('Publishing JSON results...');
-  task.addAttachment(
-    'JSON_RESULT',
-    'trivy' + Math.random() + '.json',
-    outputPath
-  );
+  task.addAttachment('JSON_RESULT', resultsFile, localOutputPath);
 
   if (hasAccount) {
     console.log('Publishing JSON assurance results...');
@@ -105,8 +105,8 @@ async function run() {
   }
 
   task.debug('Generating additional reports...');
-  if (task.exist(outputPath)) {
-    generateAdditionalReports(outputPath);
+  if (task.exist(localOutputPath)) {
+    generateAdditionalReports(localOutputPath);
   } else {
     task.error(
       'Trivy seems to have failed so no output path to generate reports from.'

--- a/trivy-task/task.json
+++ b/trivy-task/task.json
@@ -8,7 +8,7 @@
   "helpUrl": "https://github.com/aquasecurity/trivy-azure-pipelines-task",
   "category": "Test",
   "author": "Aqua Security",
-  "version": { "Major":1,"Minor":11,"Patch":32 },
+  "version": { "Major":1,"Minor":11,"Patch":36 },
   "instanceNameFormat": "Echo trivy $(version)",
   
   "groups": [

--- a/trivy-task/trivyLoader.ts
+++ b/trivy-task/trivyLoader.ts
@@ -49,14 +49,18 @@ export async function createRunner(): Promise<ToolRunner> {
   const loginDockerConfig = task.getBoolInput('loginDockerConfig', false);
   const home = homedir();
   const cwd = process.cwd();
+  const dockerHome = home + '/.docker';
+
+  // ensure the docker home dir is created
+  task.mkdirP(dockerHome);
 
   runner.line('run --rm');
   loginDockerConfig
-    ? runner.line('-v ' + task.getVariable('DOCKER_CONFIG') + ':/root/.docker')
-    : runner.line('-v ' + home + '/.docker:/root/.docker');
-  runner.line('-v /tmp:/tmp');
+    ? runner.line('-v ' + `task.getVariable('DOCKER_CONFIG') + :/root/.docker`)
+    : runner.line('-v ' + `${dockerHome}:/root/.docker`);
+  runner.line(`-v ${tmpPath}:/tmp`);
   runner.line('-v /var/run/docker.sock:/var/run/docker.sock');
-  runner.line('-v ' + cwd + ':/src');
+  runner.line(`-v ${cwd}:/src`);
   runner.line('--workdir /src');
 
   if (hasAquaAccount()) {

--- a/vss-extension.dev.json
+++ b/vss-extension.dev.json
@@ -2,7 +2,7 @@
   "manifestVersion": 1,
   "id": "trivy-official-dev",
   "publisher": "AquaSecurityOfficial",
-  "version": "1.11.32",
+  "version": "1.11.36",
   "name": "trivy-dev",
   "description": "Trivy is the world’s most popular open source vulnerability and misconfiguration scanner. It is reliable, fast, extremely easy to use, and it works wherever you need it.",
   "repository": {

